rotkehlchen/tests/api/test_defi_protocols.py

```diff
@@ -0,0 +1,532 @@
+from typing import TYPE_CHECKING
+
+import requests
+
+from rotkehlchen.tests.utils.api import api_url_for, assert_proper_response_with_result
+
+if TYPE_CHECKING:
+    from rotkehlchen.api.server import APIServer
+
+
+def test_defi_protocols_endpoint(rotkehlchen_api_server: 'APIServer') -> None:
+    """Test that the defi protocols endpoint works properly"""
+    response = requests.get(api_url_for(rotkehlchen_api_server, 'defimetadataresource'))
+
+    result = assert_proper_response_with_result(response)
+    assert result == [
+        {
+            'identifier': 'ygovfinancev1',
+            'name': 'ygov.finance (v1)',
+            'icon': 'iearn.png',
+        },
```

Comment:
instead of having this big list can't you just genereate it from `DEFI_PROTOCOLS` calling serialize?

Reply:
maybe would be better to check that all the objects have the expected structure and that the length is the same as in the provided data